velox/functions/sparksql/tests/RandTest.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class RandTest : public SparkFunctionBaseTest {
+ public:
+  RandTest() {
+    // Allow for parsing literal integers as INTEGER, not BIGINT.
+    options_.parseIntegerAsBigint = false;
+  }
+
+ protected:
+  std::optional<double> rand(int32_t seed, int32_t partitionIndex = 0) {
+    return evaluateOnce<double>(
+        fmt::format("rand({}, {})", seed, partitionIndex),
+        makeRowVector(ROW({}), 1));
+  }
+
+  std::optional<double> randWithNullSeed(int32_t partitionIndex = 0) {
+    return evaluateOnce<double>(
+        fmt::format("rand(NULL, {})", partitionIndex),
+        makeRowVector(ROW({}), 1));
+  }
+
+  std::optional<double> randWithNoSeed() {
+    return evaluateOnce<double>("rand()", makeRowVector(ROW({}), 1));
+  }
+
+  VectorPtr randWithBatchInput(int32_t seed, int32_t partitionIndex = 0) {
+    auto exprSet = compileExpression(
+        fmt::format("rand({}, {})", seed, partitionIndex), ROW({}));
+    return evaluate(*exprSet, makeRowVector(ROW({}), 20));
+  }
+
+  void checkResult(const std::optional<double>& result) {
+    EXPECT_NE(result, std::nullopt);
+    EXPECT_GE(result.value(), 0.0);
+    EXPECT_LT(result.value(), 1.0);
+  }
+
+  bool areNotEqualVectors(const VectorPtr& vector1, const VectorPtr& vector2) {
+    for (auto i = 0; i < vector1->size(); i++) {
```

Comment:
perhaps, assert that type and size of the vectors is the same, only values are different